webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java

```diff
@@ -67,27 +92,156 @@
         return taskService.getAllTasksForCurrentUser()
             .stream().map(bean -> {
                 try {
-                    bean.setSiteTitle(siteService.getSite(bean.getSiteId()).getTitle());
-                    entityManager.getUrl(bean.getReference(), Entity.UrlType.PORTAL).ifPresent(u -> bean.setUrl(u));
+                    Site site = siteService.getSite(bean.getSiteId());
+                    bean.setSiteTitle(site.getTitle());
+                    if (bean.getReference() != null && !bean.getReference().startsWith("/user/")) {
+                        entityManager.getUrl(bean.getReference(), Entity.UrlType.PORTAL).ifPresent(u -> bean.setUrl(u));	
+                    }
+                    String taskAssignedTo = getTaskAssignedDescription(bean.getTaskId(), site);
+                    bean.setTaskAssignedTo(taskAssignedTo);
                 } catch (IdUnusedException e) {
                     log.warn("No site for id {}", bean.getSiteId());
                 }
                 return bean;
             }).collect(Collectors.toList());
     }
+    
+    @GetMapping(value = "/tasks/site/{siteId}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public List<UserTaskAdapterBean> getSiteTasks(@PathVariable String siteId) throws UserNotDefinedException {
+
+        checkSakaiSession();
+
+        // Flatten the UserTask objects into a more compact form and return.
+        return taskService.getAllTasksForCurrentUserOnSite(siteId)
+            .stream().map(bean -> {
+                try {
```

Comment:
I would lift this entire getTitle bit out of the lambda. You are getting tasks for one site. Why do you need to query it again and again?